frontend/components/token-classifier/results/RecordTokenClassification.vue

```diff
@@ -195,23 +196,47 @@ export default {
           : [];
       }
     },
-    async onValidate(record) {
+    async toggleValidateRecord() {
+      if (this.record.status === "Validated") {
+        await this.onChangeStatusToDefault();
+      } else {
+        await this.onValidate();
+      }
+    },
+    async toggleDiscardRecord() {
+      if (this.record.status === "Discarded") {
+        await this.onChangeStatusToDefault();
+      } else {
+        this.onDiscard();
+      }
+    },
+    async onValidate() {
       await this.validate({
         // TODO: Move this as part of token classification dataset logic
         dataset: this.getTokenClassificationDataset(),
         agent: this.$auth.user.username,
         records: [
           {
-            ...record,
+            ...this.record,
             annotatedEntities: undefined,
             annotation: {
-              entities: record.annotatedEntities,
+              entities: this.record.annotatedEntities,
               origin: "annotation",
             },
           },
         ],
       });
     },
+    async onChangeStatusToDefault() {
+      await this.changeStatusToDefault({
+        dataset: this.getTokenClassificationDataset(),
+        records: [
+          {
+            ...this.record,
+          },
+        ],
+      });
+    },
```

Comment:
1/ add a factory function before passing it to the `changeStatusToDefault` function
2/ the destructuration `{... this.record}` seems better to be replaced directly by `this.record`